authz/audit_logger.go

```diff
@@ -0,0 +1,109 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package authz
+
+import (
+	"context"
+	"sync"
+)
+
+var (
+	mu sync.Mutex
+	m  = make(map[string]AuditLoggerBuilder)
```

Comment:
I am not aware of any convention. This is for consistency within this repository. For example, the map in the `balancer` pkg is named `m`.

Reply:
Personally I'd be a fan of giving it at least some kind of meaningful name unless that is a very strong convention

Reply:
I added a comment on this var following the `balancer` pkg. I still think consistency is more beneficial in that if someone needed to do something similar in the future, they wouldn't need to spend extra time thinking which prior art to follow. Let me know if you don't think this is good enough. ;)

Reply:
This being in the `authz` pkg is a strong point that I didn't realize. I changed the name to `auditLoggerBuilderRegistry`.

Reply:
Well now it feels like Java...

Reply:
I renamed it to `loggerBuilderRegistry`. Hopefully it's less like Java now ;-)

Reply:
Was a separate package for audit logging inside the `authz` directory considered? If so, what were the reasons for not taking that approach? I feel that it might be cleaner to have an `auditlog` package inside the `authz` directory, since audit logger is a reasonably separate component in itself. 

Also, I don't see the audit logging functionality depend on anything in the `authz` package. So, I believe circular dependencies might not be an issue here.